digital_image_processing/index_calculation.py

```diff
@@ -1,651 +1,525 @@
-        return (max - min) / max
-
-    def IF(self, red=None, green=None, blue=None):
-        """
-            Shape Index
-            https://www.indexdatabase.de/db/i-single.php?id=79
-            :return: index
-        """
-        return (2 * red - green - blue) / (green - blue)
-
-    def DVI(self, red=None, nir=None):
-        """
-            Simple Ratio NIR/RED Difference Vegetation Index, Vegetation Index Number (VIN)
-            https://www.indexdatabase.de/db/i-single.php?id=12
-            :return: index
-        """
-        return nir / red
-
-    def TVI(self, red=None, nir=None):
-        """
-            Transformed Vegetation Index
-            https://www.indexdatabase.de/db/i-single.php?id=98
-            :return: index
-        """
-        return (self.NDVI(red=red, nir=nir) + 0.5) ** (1 / 2)
-
-    def NDRE(self, redEdge=None, nir=None):
-        return (nir - redEdge) / (nir + redEdge)
-
-
-"""
-# genering a random matrices to test this class
-red     = np.ones((1000,1000, 1),dtype="float64") * 46787
-green   = np.ones((1000,1000, 1),dtype="float64") * 23487
-blue    = np.ones((1000,1000, 1),dtype="float64") * 14578
-redEdge = np.ones((1000,1000, 1),dtype="float64") * 51045
-nir     = np.ones((1000,1000, 1),dtype="float64") * 52200
-
-# Examples of how to use the class
-
-# instantiating the class 
-cl = IndexCalculation()
-
-# instantiating the class with the values
-#cl = indexCalculation(red=red, green=green, blue=blue, redEdge=redEdge, nir=nir)
-
-# how set the values after instantiate the class cl, (for update the data or when dont instantiating the class with the values)
-cl.setMatrices(red=red, green=green, blue=blue, redEdge=redEdge, nir=nir)
-
-# calculating the indices for the instantiated values in the class
-indexValue    = cl.calculation("CCCI").astype(np.float64)
-
-# calculating the index with the values directly 
-#indexValue    = cl.directCalculation("CCCI",red=red, green=green, blue=blue, redEdge=redEdge, nir=nir).astype(np.float64)
-
-
-print(np.array2string(indexValue, precision=20, separator=', ', floatmode='maxprec_equal'))
-# A list of examples results for different type of data at 'calculation' or 'directCalculation' 
-# float16 ->    0.31567383              #NDVI (red = 50, nir = 100)
-# float32 ->    0.31578946              #NDVI (red = 50, nir = 100)
-# float64 ->    0.3157894736842105      #NDVI (red = 50, nir = 100)
-# longdouble -> 0.3157894736842105      #NDVI (red = 50, nir = 100)
-"""
+# Author: João Gustavo A. Amorim
+# Author email: joaogustavoamorim@gmail.com
+# Coding date:  jan 2019
+# python/black: True
+
+# Imports
+import numpy as np
+
+# Class implemented to calculus the index
+class IndexCalculation:
+    """
+        # Class Summary
+            This algorithm consists in calculating vegetation indices, these indices can be used for precision agriculture for example (or remote sensing). There are functions to define the data and to calculate the implemented indices.
```

Comment:
These lines should be wrapped so that they can be read on GitHub with left-right scrolling.